stan/math/opencl/tri_inverse.hpp

```diff
@@ -81,7 +81,7 @@ inline matrix_cl<T> tri_inverse(const matrix_cl<T>& A) {
   zero_mat.template zeros<stan::math::matrix_cl_view::Entire>();
   inv_padded.template zeros<stan::math::matrix_cl_view::Entire>();
   if (tri_view == matrix_cl_view::Upper) {
-    inv_mat = transpose(inv_mat);
+    inv_mat = transpose(inv_mat).eval();
```

Comment:
Why is there a need for `.eval()` here? Shouldn't the kernel kick off once it's being assigned to an already constructed `matrix_cl`?

Reply:
The problem is that without eval we have aliasing issues, similar as [Eigen has](https://eigen.tuxfamily.org/dox/group__TopicAliasing.html). Since source and destination is the same matrix threads that work on lower/upper triangular part can (and do) overwrite each other's input values with their outputs. With eval we create new matrix for destination. That is also how individual transpose kernel works.

Reply:
Okay if you can file an issue to write some docs about this (and then add the docs later) then I'm cool with approving this. We should probably have something like Eigen does if we also have the same aliasing issues.

You can either add that as a module (like in the below comment) or follow the instructions [here](https://github.com/stan-dev/math/blob/develop/doxygen/howto_add_page.md) on adding a new page